apriltags_ros/src/apriltag_detector.cpp

```diff
@@ -456,40 +457,101 @@ tf::Transform AprilTagDetector::getDepthImagePlaneTransform(const sensor_msgs::P
 
   pcl::PointIndices::Ptr polygonInlierIndices(new pcl::PointIndices());
 
-  if(!same_frame_id_)
-  {
-    ROS_DEBUG_THROTTLE(5.0, "Register polygon points from rgb to depth frame");
-    for(size_t i = 0; i < 4; i++)
-    {
-      polygon[i].first = (polygon[i].first - rgb_info->K[2]) / rgb_info->K[0] * depth_info->K[0] + depth_info->K[2];
-      polygon[i].second = (polygon[i].second - rgb_info->K[5]) / rgb_info->K[4] * depth_info->K[4] + depth_info->K[5];
-    }
-  }
-
   pcl::PointCloud<pcl::PointXYZ> clipPolygon;
   clipPolygon.push_back(pcl::PointXYZ(polygon[0].first, polygon[0].second, 0));
   clipPolygon.push_back(pcl::PointXYZ(polygon[1].first, polygon[1].second, 0));
   clipPolygon.push_back(pcl::PointXYZ(polygon[2].first, polygon[2].second, 0));
   clipPolygon.push_back(pcl::PointXYZ(polygon[3].first, polygon[3].second, 0));
   clipPolygon.push_back(pcl::PointXYZ(polygon[0].first, polygon[0].second, 0));
 
-  for (int x = 0; x < pointCloud->width; x++)
-  {
-    for (int y = 0; y < pointCloud->height; y++)
-    {
-        int i = x + (pointCloud->width * y);
+  // generate the transform from depth_camera_reference frame to rgb_camera_reference frame
+  tf::Transform depth_to_rgb_transform;
+  string rgb_camera_frame_name = rgb_info->header.frame_id;
+  string depth_camera_frame_name = cloud->header.frame_id;
+
+  pcl::PointCloud<pcl::PointXYZRGB> point_cloud_in_rgb_frame;
+
+  depth_to_rgb_transform.setIdentity();
+  depth_to_rgb_transform.setOrigin(tf::Vector3{depth_info->P[3],depth_info->P[7],depth_info->P[11]});
```

Comment:
Why are we using depth intrinsics to convert the depth to rgb frame

Reply:
Because by default camera 0 (rgb) has offset of 0. The depth camera has the offset Tx, Ty, Tz non zero with respect to the rgb camera. These offsets are such that B (actual distance between cameras) = Tx/-fx.